### PR TITLE
support asset paths for annotation library sources

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ dependencies:
   args: '>=0.12.2+3 <0.14.0'
   cli_util: ^0.0.1
   dart_style: '>=0.1.7 <0.3.0'
+  mockito: ^0.11.0
```
Comment (kevmoo) [inline pubspec.yaml]: make this a dev_dependency

File: test/annotation_test.dart [hunk 2]
```diff
@@ -221,12 +224,25 @@ void main() {
...
+        test('When running on a normal file system', () {
+          ClassElement annotatedClass =
+              _getAnnotationForClass(libElement, 'AnnotatedWithJson');
+          var annotation = annotatedClass.metadata.single;
+          var matched = matchAnnotation(JsonSerializable, annotation);
+          expect(matched, isTrue);
+        });
+
+        test('When running in barback', () {
+          var annotation = _mockElementAnnotation(
```
Comment (kevmoo) [inline test/annotation_test.dart]: Is there anyway to do this without mock silly? I worry about this being brittle...

Comment (jakemac53) [inline test/annotation_test.dart]: Ya, its definitely brittle. All the other ways I can think of to test this are a lot worse though, and would require a barback dependency and mocking out a bunch of barback stuff :/

**Comment by kevmoo** [comment]
LGTM

**Pull request merged**